File: Transfer Matrix/Multilayer/Material.py
import numpy as np
import materials

MATERIALIDS = {
    0:materials.Air,
    1:materials.SiO2,
    2:materials.Si
}

def Material(materialid,frequency):
    N = len(materialid)
    M = len(frequency)
    epsilon = np.ones([N,M])
    mur = np.ones([N,M])
    MaxType = len(MATERIALIDS)

    for n in range(N):
        if materialid[n] < MaxType:
            [eps,mu] = MATERIALIDS[materialid[n]](frequency)
            epsilon[n,:] = eps
            mur[n,:] = mu

    return np.array([epsilon,mur])

File: Transfer Matrix/julia/material.jl
include("./materials/air.jl")
include("./materials/si.jl")
include("./materials/sio2.jl")

MaterialIDs = Dict(
    0 => Air,
    1 => SiO2,
    2 => Si
)

function Material(materialID, frequency)
    N = length(materialID)
    M = length(frequency)
    epsilons = ones((N, M))
    mus = ones((N, M))
    maxtype = length(MaterialIDs)

    for i in 1:N
        if materialID[i] <= maxtype
            eps, mu = MaterialIDs[materialID[i]](frequency)
            epsilons[i, :] = eps
            mus[i, :] = mu
        end
    end

    [epsilons, mus]
end